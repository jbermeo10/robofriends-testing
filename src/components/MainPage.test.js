import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

// it('se espera renderizar el componente MainPage', () => {
//   const mockStore = {
//     robots: [],
//     searchField: ''
//   }
//   expect(shallow(<MainPage store={mockStore}/>)).toMatchSnapshot();
// })

let wrapper;

beforeEach( () =>  {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false,
    error: ''
  }
  wrapper = shallow(<MainPage { ...mockProps}/>)
})


it('renderiza MainPage sin caerse', () => {
  expect(wrapper).toMatchSnapshot();
})

it('filtra los robots correctamente', () => {
  expect(wrapper.instance().filterRobots()).toEqual([]);
})

it('filtra los robots correctamente 2', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'john',
    isPending: false,
    error: ''
  }
  const wrapper2 = shallow(<MainPage { ...mockProps2}/>)
  expect(wrapper2.instance().filterRobots()).toEqual([{
    id: 3,
    name: 'John',
    email: 'john@gmail.com'
  }])
})

it('filtra los robots correctamente 3', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: 'John',
      email: 'john@gmail.com'
    }],
    searchField: 'a',
    isPending: false,
    error: ''
  }
  const filteredRobots = []
  const wrapper3 = shallow(<MainPage { ...mockProps3}/>)
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots)
})