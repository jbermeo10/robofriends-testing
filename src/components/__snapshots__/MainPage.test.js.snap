// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza MainPage sin caerse 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
